### PR TITLE
refactor(types): rename locale definition types to singular

### PR DESCRIPTION
**Pull request by ST-DDT**
Rename:

- `LocaleDefinition` -> `LocaleDefinitions`
- `LocaleEntry` -> `LocaleCategoryDefinitions` (cause it is only for categories)

I'm not sure about the new names and whether we should use singular or plural names for these.

### CODE REVIEW

**Review by xDivisionByZerox — DISMISSED** [review]

**Comment by Shinigami92** [comment]
Especially `locale: LocaleDefinitions | LocaleDefinitions[]` is a good indicator to me that we should use singular (I'm sorry :shrug:)

**Comment by ST-DDT** [comment]
We should decide this before releasing v8.0

**Comment by Shinigami92** [comment]
> We should decide this before releasing v8.0

I vote for singular

**Comment by Shinigami92** [comment]
@xDivisionByZerox What do you think is the better option?

**Comment by xDivisionByZerox** [comment]
I prefer the singular variant (`LocaleDefinition`, `LocaleCategoryDefinition`). 

You could say that the object, represented by the type, is a collection of keys - similar to a `Map`. 
While you would most likely write the value of a `Map` in the plural form (`const books = new Map();`), I don't see the argument being fitting for our specific case. A locale definition may be a collection of suggested definitions but it still applies to one locale only (so singular).

Thats at least how I see it.

**Comment by ST-DDT** [comment]
Changed to singular

**Comment by matthewmayer** [comment]
If this is considered a breaking change then it should be mentioned in migration guide.

**Comment by ST-DDT** [comment]
> If this is considered a breaking change then it should be mentioned in migration guide.

No, it is not because the old aliases are still exported.

**Review by Shinigami92 — DISMISSED** [review]
LGTM

File: src/definitions/definitions.ts [hunk 1]
```diff
@@ -28,28 +28,26 @@ export type LocaleEntry<T extends Record<string, unknown>> = {
...
+  location?: LocationDefinition;
+  lorem?: LoremDefinition;
+  music?: MusicDefinition;
+  person?: PersonDefinition;
+  phone_number?: PhoneNumberDefinition;
+  science?: ScienceDefinition;
+  system?: SystemDefinition;
+  vehicle?: VehicleDefinition;
+  word?: WordDefinition;
+} & Record<string, Record<string, unknown>>;
```
Comment (xDivisionByZerox) [inline src/definitions/definitions.ts]: Why did you remove the `| undefined` from the custom modules?

Comment (ST-DDT) [reply src/definitions/definitions.ts]: This was not intentional. I noticed an error showing up in the locale-proxy tests that is caused by that `undefined` and tried addressing this but not as part of this PR and more for #1834 / #2102.

Reverted

**Review by Shinigami92 — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]
